### PR TITLE
Map demo + Guide

### PR DESCRIPTION
**Pull request by dawoodkhan82**
# Description
Created map demo with plotly, and an associated guide for it.

https://huggingface.co/spaces/dawood/NYC-Airbnb-Map

Please include: 
* relevant motivation
* a summary of the change 
* which issue is fixed. 
* any additional dependencies that are required for this change.

Closes: #2338, #1993 

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have added a short summary of my change to the CHANGELOG.md
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md, unless the pull request is labeled with the "no-changelog-update" label.

Please add a brief summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

**Review by freddyaboulton — APPROVED** [review]
Awesome demo @dawoodkhan82 ! 🗺️ 

I gave a sneak peek of this demo at the talk yesterday and people were into it. 

I think there's a small bug in the tooltip text. Might be good to add this to the demos tab as well.

**Review by freddyaboulton — COMMENTED** [review]
If we're worried about the size of the csv we can host on Hub datasets as well.

File: guides/5)other_tutorials/plot_component_for_maps.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+In the code above, we first load the csv data into a pandas dataframe. Let's begin by defining a function that we will use as the prediction fuction for the gradio app. This function will accept the minimum price and maximum price range as well as the list of boroughs to filter the resulting map. We can use the passed in values (`min_price`, `max_price`, and list of `boroughs`) to filter the dataframe and create `new_df`. Next we will create `text_list` of the names and prices of each Airbnb to use as labels on the map.
+ 
+## Step 2 - Map Figure 🌐
+
+Plotly makes it easy to work with maps. Let's take a look below how we can create a map figure.
+
+```python
+import plotly.graph_objects as go
+
+fig = go.Figure(go.Scattermapbox(
```
[the remark this answers is not in the document]
Comment (dawoodkhan82) [reply guides/5)other_tutorials/plot_component_for_maps.md:44]: correct. using open street maps

File: guides/5)other_tutorials/plot_component_for_maps.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+## Step 1 - Loading CSV data 💾
+
+Let's start by loading the Airbnb NYC data from the csv data we [downloaded](https://www.kaggle.com/datasets/dgomonov/new-york-city-airbnb-open-data).
+
+```python
+import pandas as pd
+
+df = pd.read_csv("AB_NYC_2019.csv")
+
+def filter_map(min_price, max_price, boroughs):
```
[the remark this answers is not in the document]
Comment (dawoodkhan82) [reply guides/5)other_tutorials/plot_component_for_maps.md:27]: oh yeah good catch!

File: guides/5)other_tutorials/plot_component_for_maps.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        ),
+        pitch=0,
+        zoom=9
+    ),
+)
+```
+
+Above, we create a scatter plot on mapbox by passing it our list of latitudes and longitudes to plot markers.  We also pass in our custom data of names and prices for additional info to appear on every marker we hover over. Next we use `update_layout` to specify other map settings such as zoom, and centering.
+
+More info [here](https://plotly.com/python/scattermapbox/) on scatter plots using Mapbox and Plotly.
```
Comment (abidlabs) [inline guides/5)other_tutorials/plot_component_for_maps.md:73]: It would be good to describe what other map types besides `Scattermapbox` are supported.

**Comment by abidlabs** [comment]
> If we're worried about the size of the csv we can host on Hub datasets as well.

Agreed, the CSV is rather large. You could even just include the python code to download the CSV file from the Kaggle link using the `requests` library as part of `run.py` and as part of the Guide.

**Comment by abidlabs** [comment]
Otherwise, the Guide reads great! Nice work @dawoodkhan82

**Review by abidlabs — APPROVED** [review]

**Pull request merged**